apps/system/utils/kdbg_date.c

```diff
@@ -153,10 +161,23 @@ static inline int date_settime(int argc, char **args)
 	long result;
 	int ret = OK;
 
+	/* Get the year */
+
+	token = args[5];
+	if (token == NULL) {
+		goto errout_bad_parm;
+	}
+
+	result = strtol(token, NULL, 10);
+	if (result < TM_YEAR_BASE || result > 2100) {
```

Comment:
the 1st check should be `result < 1970`. Setting year which is less than `1970` (**EPOCH YEAR**) is wrong

Reply:
@lokeshbv When I changed calculating logic of time, I got correct answer.
```
TASH>>date -s Feb 10 11:12:13 1932
mktime -1195735667
TASH>>date
show -1195735666
Feb 10 11:12:14 1932
```
Standard year for epoch is 1970. But time_t type is long so that it can be a negative.
There is no reason not to support less than 1970.